libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/LoginHost/SFSDKLoginHostStorage.m

```diff
@@ -60,19 +60,23 @@ + (SFSDKLoginHostStorage *)sharedInstance {
 
 - (id)init {
     self = [super init];
-    
     if (self) {
         self.loginHostList = [NSMutableArray array];
-        
-        // Add the Production and Sandbox login hosts are defined. These two items cannot be deleted.
-        [self.loginHostList addObject:[SFSDKLoginHost hostWithName:[SFSDKResourceUtils localizedString:@"LOGIN_SERVER_PRODUCTION"] host:@"login.salesforce.com" deletable:NO]];
-        
-        [self.loginHostList addObject:[SFSDKLoginHost hostWithName:[SFSDKResourceUtils localizedString:@"LOGIN_SERVER_SANDBOX"] host:@"test.salesforce.com" deletable:NO]];
-        
-        // Load from managed preferences (e.g. MDM).
         SFManagedPreferences *managedPreferences = [SFManagedPreferences sharedPreferences];
+
+        // Add the Production and Sandbox login hosts, unless an MDM policy explicitly forbids this.
+        if (!(managedPreferences.hasManagedPreferences && managedPreferences.onlyShowAuthorizedHosts)) {
+            [self.loginHostList addObject:[SFSDKLoginHost hostWithName:[SFSDKResourceUtils localizedString:@"LOGIN_SERVER_PRODUCTION"] host:@"login.salesforce.com" deletable:NO]];
+            [self.loginHostList addObject:[SFSDKLoginHost hostWithName:[SFSDKResourceUtils localizedString:@"LOGIN_SERVER_SANDBOX"] host:@"test.salesforce.com" deletable:NO]];
+        }
+
+        // Load from managed preferences (e.g. MDM).
         if (managedPreferences.hasManagedPreferences) {
-            //If ther are any existing login hosts, remove them as MDM should take highest priority and only the hosts enforced by MDM should be in the list.
+
+            /*
+             * If there are any existing login hosts, remove them as MDM should take
+             * highest priority and only the hosts enforced by MDM should be in the list.
+             */
             if([self.loginHostList count] > 0) {
```

Comment:
If MDM is used for an configuration, it also has to be used to provide login hosts?